config/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,115 @@
+package v1alpha2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsDataGather provides data gather configuration options for the the Insights Operator.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=insightsdatagathers,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1245
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=config-operator,operatorOrdering=01
+// +openshift:enable:FeatureGate=InsightsConfig
+// +openshift:compatibility-gen:level=4
+type InsightsDataGather struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// spec holds user settable values for configuration
+	// +required
+	Spec InsightsDataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsDataGatherStatus `json:"status"`
+}
+
+type InsightsDataGatherSpec struct {
+	// gatherConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights data and its uploading to the ingress.
+	// +optional
+	GatherConfig GatherConfig `json:"gatherConfig,omitempty"`
```

Comment:
Did we ever work out if we needed any siblings to this field? I wonder if we could promote everything below this up one level?

Reply:
So if I understand it correctly you suggest using something like this?
```go
type InsightsDataGatherSpec struct {
	DataPolicy DataPolicy       `json:"dataPolicy,omitempty"`
	Gatherers  []GathererConfig `json:"gatherers,omitempty"`
}
```
I don't know about any other fields that should be added at the moment, so I would drop that extra level. WDYT @tremes @ncaak

Reply:
Hi there,

I think the original plan was having the configuration out of possible new uses for the type.

@JoelSpeed what are the advantages of changing this structure from the previous v1alpha1 version ?

Reply:
Just one less level of indentation to be honest
```
spec:
  gatherConfig:
    dataPolicy: ...
    gatherers: ...
```
vs
```
spec:
  dataPolicy: ...
  gatherers: ...
```

If we don't think we will ever add a sibling alongside `gatherConfig`, then separating these into their own struct has little value, but if we might add more in the future, then happy to leave as is

It strikes me though, what would a resource do if it were just:
```
spec: {}
```

Should we require at least one property under spec to be present?

Reply:
We are not sure if we will add anything in the future, so let's leave it as it is